server/utils/mapYoutubeResult.ts

```diff
@@ -0,0 +1,27 @@
+export interface YoutubeResult {
+  id: {
+    videoId: string
+  }
+  snippet: {
+    title: string
+    thumbnails: {
+      default: {
+        url: string
+      }
+    }
+  }
+}
+
+export interface YoutubeVideo {
+  url: string
+  name: string
+  img: string
+}
+
+export const mapYoutubeResult = (video: YoutubeResult): YoutubeVideo => {
```

Comment:
I kinda like having all the type declarations in one file and move the map function to the "functions" youtube file.